### PR TITLE
ENT-7544: Added new app academy with related models

### PR DESCRIPTION
**Pull request by saleem-latif**
## Description
An Academy backend is composed of multiple catalogs of learning content mapped as a single level tree of nodes. Each node stores an enterprise catalog as a definition of content included in the node.
An academy will be associated with content metadata similar to how an enterprise is associated with content metadata.

## Ticket Link
[Academies app in enterprise catalog](https://openedx.atlassian.net/browse/ENT-7544)

## Post-review

* [ ] Squash commits into discrete sets of changes
* [ ] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

File: enterprise_catalog/apps/academy/models.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Academy(TimeStampedModel):
+    """
+    Model for storing academy related information.
+
+    .. no_pii:
+    """
+    uuid = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    title = models.CharField(max_length=255, help_text=_('Academy title'))
+    enterprise_catalogs = models.ManyToManyField(EnterpriseCatalog)
```
Comment (irfanuddinahmad) [inline enterprise_catalog/apps/academy/models.py]: We should add long and short description.

Comment (irfanuddinahmad) [inline enterprise_catalog/apps/academy/models.py]: Please add a related_name for the reverse relationship e.g.
`enterprise_catalogs = models.ManyToManyField(EnterpriseCatalog, related_name='academies')`

**Review by irfanuddinahmad — CHANGES_REQUESTED** [review]

**Review by irfanuddinahmad — APPROVED** [review]
LGTM!

**Review by muhammad-ammar — APPROVED** [review]
LGTM

**Pull request merged**